RUFAS/data_collection_app_updater.py

```diff
@@ -429,3 +443,112 @@ def _add_filename_input_field(self, schema: dict[str, Any]) -> dict[str, Any]:
+            file.write(updated_js_content)
+
+    def modify_items_schema(
+        self, data: dict[str, Any], dropdown_data: dict[str, list[Any]], skip_first: bool = True
+    ) -> dict[str, Any]:
+        """
+        Modify the schema with dropdowns by updating the corresponding field with updated feed data.
+
+        Parameters
+        ----------
+        data : dict[str, Any]
+            The schema structure.
+        dropdown_data : dict[str, list]
+            The updated content in the dropdown.
+        skip_first : bool, default=True
+            Boolean indicators to help skip the first "properties" field
+
+        """
+        new_data = copy.deepcopy(data)
+        if isinstance(new_data, dict):
+            if "properties" in new_data:
+                if skip_first:
+                    skip_first = False
+                else:
+                    for key, value in new_data["properties"].items():
+                        if isinstance(value, dict):
+                            new_data["properties"][key] = self.modify_items_schema(value, dropdown_data, skip_first)
+
+            if "items" in new_data and isinstance(new_data["items"], dict):
+                items_data = new_data.get("items", {})
+
+                if "properties" in items_data and isinstance(items_data["properties"], dict):
+                    items_data["properties"] = self.update_first_property_with_enum(
+                        items_data["properties"], dropdown_data
+                    )
+
+                else:
+                    items_data["enum"] = dropdown_data["id"]
+                    if "options" not in items_data:
+                        items_data["options"] = {}
+                    items_data["options"]["enum_titles"] = dropdown_data["name"]
+
+            for key, value in new_data.items():
+                new_data[key] = self.modify_items_schema(value, dropdown_data, skip_first)
+            return new_data
+        else:
+            return new_data
+
+    @staticmethod
+    def update_first_property_with_enum(properties: dict[Any, Any], dropdown_data: dict[str, Any]) -> dict[Any, Any]:
+        """
+        Create a new properties dictionary with the first dictionary property.
+
+        Parameters
+        ----------
+        properties : dict[Any, Any]
+            The target properties to update with dropdown menu data.
+        dropdown_data : dict[str, Any]
+            The data containing the dropdown values to insert.
+
+        Returns
+        -------
+        dict[Any, Any]
+            A new dictionary with the update applied.
+
+        """
+        new_properties = copy.deepcopy(properties)
+        for key, prop_value in new_properties.items():
+            if isinstance(prop_value, dict):
+                prop_value["enum"] = dropdown_data["id"]
+                prop_value["options"] = {**prop_value.get("options", {}), "enum_titles": dropdown_data["name"]}
+                break
+
+        return new_properties
```

Comment:
```suggestion
        for prop_value in properties.values():
            if isinstance(prop_value, dict):
                prop_value["enum"] = dropdown_data["id"]
                prop_value["options"] = {**prop_value.get("options", {}), "enum_titles": dropdown_data["name"]}
                break

        return properties
```
The `deepcopy` isn't necessary (as far as I can tell) and should be avoided if at all possible because it is very resource intensive. Same for the deepcopy in `modify_items_schema`